functions/go/templater/README.md

```diff
@@ -0,0 +1,46 @@
+# Templater function
+
+This is an example of implementing a templater function.
```

Comment:
what's a "templater". This paragraph doesn't actually explain what this does.

Reply:
I've updated the text below

Reply:
Thanks for the clarification. Even though we CAN create a templating function like this, it goes against the Config as Data principles:

https://googlecontainertools.github.io/kpt/concepts/architecture/#configuration-as-data

Which is an important motivator for why Kpt/Kustomize projects exist.

Reply:
I don't think this is resolved. Nothing preventing users from using a templater function, but having it in this catalog is against the design philosophies of Config as Data

Reply:
oh.. ok, It seems I missed the last comment and was just waiting... I'll look into that and will try to address it. Sorry for the delay.

Reply:
@frankfarzan 
I've tried to modify the function, so it could employ go-template engine + sprig as a scripting lang to perform transformation.
To be able to work with yaml-resources, I've introduced [KYAML go-template library](https://github.com/aodinokov/kpt-functions-catalog/tree/function-with-env/functions/go/templater#kyaml-go-template-library). The new version of README contains some [examples](https://github.com/aodinokov/kpt-functions-catalog/tree/function-with-env/functions/go/templater#kyaml-go-template-library-usage-examples) how this function can be used.
The result is pretty universal and powerful. Hope you'll like this concept, since it's based on kyaml primitives. 
@monopole asked to create a simple function to be able to test that env variables pass. Even though this function isn't absolutely simple, it's still pretty short for the functionality it can provide. I think this function can be used for a big amount of different cases: filtering, annotating/lableling/modification of resources. It may be useful  especially for non-standard cases.

Let me know if it's too complicated or you have other concerns.
If it's not something you want to support I can make something really simple for the testing purposes of https://github.com/kubernetes-sigs/kustomize/pull/2999 . I just wanted to make something meaningful and useful, not just yet-another-example of set-label fn, that takes env var for that :)